### PR TITLE
Integrate augeas project to oss-fuzz.

### PR DESCRIPTION
**Pull request by qiankehan**

### CODE REVIEW

**Comment by qiankehan** [comment]
I signed it!

File: projects/augeas/augeas_escape_name_fuzzer.cc [hunk 1]
```diff
@@ -0,0 +1,43 @@
+/*
```
Comment (kcc) [inline projects/augeas/augeas_escape_name_fuzzer.cc]: Any chance to have this fuzz target in the upstream repository?

File: projects/augeas/project.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+homepage: https://github.com/hercules-team/augeas
+primary_contact: lutter@watzmann.net
```
Comment (kcc) [inline projects/augeas/project.yaml]: We will need one of the main project owners (as per the github project) to agree with you doing this.

File: projects/augeas/build.sh [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+./autogen.sh
+CC=clang ./configure --enable-static --disable-shared --without-selinux
+make -j$(nproc)
+
+for fuzzer in augeas_escape_name_fuzzer; do
```
Comment (kcc) [inline projects/augeas/build.sh]: This is ok as the first step, but we recommend to have the fuzz targets be tighter integrated into the project's build system. See https://github.com/google/oss-fuzz/blob/master/docs/ideal_integration.md

File: projects/augeas/build.sh [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+./autogen.sh
+CC=clang ./configure --enable-static --disable-shared --without-selinux
```
Comment (kcc) [inline projects/augeas/build.sh]: Why do you need `CC=clang` here?

Comment (qiankehan) [reply projects/augeas/build.sh]: Well, it is unnecessary here. I will remove it.

File: projects/augeas/project.yaml (hunk 2 above)
Comment (lutter) [reply projects/augeas/project.yaml]: Hi, yes, I totally agree with this, and would love to have augeas added to oss-fuzz. Please let me know if this statement is enough or if I have to sign up somewhere else.

Comment (kcc) [reply projects/augeas/project.yaml]: @lutter Yes, this comment is sufficient, thanks!

File: projects/augeas/augeas_escape_name_fuzzer.cc (hunk 1 above)
Comment (qiankehan) [reply projects/augeas/augeas_escape_name_fuzzer.cc]: I'd like to start the fuzz target here first and later I will add it in upstream repository.

Comment (kcc) [reply projects/augeas/augeas_escape_name_fuzzer.cc]: ok!

**Pull request merged**